scripts/data_analysis/constants/common.py

```diff
@@ -25,3 +35,47 @@ def extract_data(url=URL):
     else:
         print(url, " responded with ", response.status_code)
         return '{}'
+
+def findParentSchema(response_dict, schema_name):
+    """
+    Search all schemas for any reference to target schema (i.e. find all direct parents) and return results in a dataframe.
+    This function assumes the dictionary object passed in is generated from OpenAPI Swagger spec
+
+        Parameters:
+            response_dict (dict): A dictionary object representing the extract from API Docs, such as the output of the extract_data function above
+            schema_name (str): A string of the exact schema name to search for
+
+        Returns:
+            df (pd.DataFrame): a dataframe of schema, field and reference information.
```

Comment:
Personally, not a fan of variables that are acronyms. Would prefer `data_frame` and `data_frames` especially helpful a not very experienced Python person. However I'm also on the fence as I've seen this kind of pattern in Python befoar! 😅

Reply:
It's a tricky one I know what you mean, as very well accepted nomenclature for a dataframe is 'df', however it does break the rule of not being easily understood to all. I will change to data_frame to make it more explicit